jsapp/js/account/usage.component.tsx

```diff
@@ -1,89 +1,237 @@
-import React, {useEffect, useState} from 'react';
+import {when} from 'mobx';
+import React, {useEffect, useMemo, useState} from 'react';
+import {useLocation} from 'react-router-dom';
+import type {AccountLimit, RecurringInterval} from 'js/account/stripe.api';
+import {getAccountLimits, getSubscriptionInterval} from 'js/account/stripe.api';
+import subscriptionStore from 'js/account/subscriptionStore';
+import LoadingSpinner from 'js/components/common/loadingSpinner';
+import UsageContainer from 'js/components/usageContainer';
+import envStore from 'js/envStore';
+import {formatDate} from 'js/utils';
+import {getUsageForOrganization} from './usage.api';
 import styles from './usage.module.scss';
-import {formatMonth} from '../utils';
-import {getUsage} from './usage.api';
-import Icon from '../components/common/icon';
-import {NavLink} from 'react-router-dom';
-import Button from '../components/common/button';
 
 interface UsageState {
   storage: number;
-  monthlySubmissions: number;
-  monthlyTranscriptionMinutes: number;
-  monthlyTranslationChars: number;
+  submissions: number;
+  transcriptionMinutes: number;
+  translationChars: number;
+  storageByteLimit: number | 'unlimited';
+  nlpCharacterLimit: number | 'unlimited';
+  nlpMinuteLimit: number | 'unlimited';
+  submissionLimit: number | 'unlimited';
+  trackingPeriod: RecurringInterval;
+  currentMonthStart: string;
+  currentYearStart: string;
+  isUsageLoaded: boolean;
+  isLimitsLoaded: boolean;
+  isSubscriptionLoaded: boolean;
 }
 
 export default function Usage() {
   const [usage, setUsage] = useState<UsageState>({
     storage: 0,
-    monthlySubmissions: 0,
-    monthlyTranscriptionMinutes: 0,
-    monthlyTranslationChars: 0,
+    submissions: 0,
+    transcriptionMinutes: 0,
+    translationChars: 0,
+    storageByteLimit: 'unlimited',
+    nlpCharacterLimit: 'unlimited',
+    nlpMinuteLimit: 'unlimited',
+    submissionLimit: 'unlimited',
+    trackingPeriod: 'month',
+    currentMonthStart: '',
+    currentYearStart: '',
+    isUsageLoaded: false,
+    isLimitsLoaded: false,
+    isSubscriptionLoaded: false,
   });
 
-  function truncate(decimal: number) {
-    return parseFloat(decimal.toFixed(2));
-  }
+  const location = useLocation();
+
+  const truncate = (decimal: number) => parseFloat(decimal.toFixed(2));
```

Comment:
We have identical function in `usageCalculations.ts`. Let's deduplicated and move to `utils.ts` :)